src/metabase/pulse.clj

```diff
@@ -209,26 +224,45 @@
         (str "…"))
     mrkdwn))
 
+(defn- text->markdown-block
+  [text]
+  (let [mrkdwn (markdown/process-markdown text :slack)]
+    (when (not (str/blank? mrkdwn))
+      {:blocks [{:type "section"
+                 :text {:type "mrkdwn"
+                        :text (truncate-mrkdwn mrkdwn block-text-length-limit)}}]})))
+
+(defn- content->attachment-data
+  [content channel-id]
+  (let [{ttype                                  :type
+         {card-id :id card-name :name :as card} :card
+         dashcard                               :dashcard
+         result                                 :result
+         tab-title                              :tab-title} content]
```

Comment:
Most of this destructuring is specific to the card type and would ideally happen inside a new let statement on line 244https://github.com/metabase/metabase/pull/30478/files#diff-a780e860575d9afab3850662e50f677408d861f5759cc91c17e985f023e7fed4R244